docs/cudf/source/conf.py

```diff
@@ -126,7 +131,13 @@ def clean_all_xml_files(path):
     for fn in glob.glob(os.path.join(path, "*.xml")):
         tree = ET.parse(fn)
         clean_definitions(tree.getroot())
-        tree.write(fn)
+        with tempfile.NamedTemporaryFile() as tmp_fn:
+            tree.write(tmp_fn.name)
```

Comment:
How does this save time? How is it faster to write to a tempfile, compare to `fn`, and then write `fn` versus just writing `fn`?

Reply:
My guess is that this particular part is _slower_, but sphinx probably has some make-like timestamp-based "do I rebuild dependents of this file" logic. So by avoiding updating a file with a newer timestamp and the same contents, something downstream does less work.

Reply:
Correct. This change no longer has a meaningful impact in PR CI because we don't build text docs, but in branch/nightly builds or local rebuilds it will make things faster because without this Sphinx tries to read the entire set of files again thinking that they have changed. Instead of calling `tree.write` here I could also have used os.rename/shutil.move etc but the amount of time taken by this part of the code is negligible and I didn't feel like dealing with issues around manual clean up of the temp file etc.